frontend/src/ui/wallet/shared/wallet-operations-wrapper-view.js

```diff
@@ -1,52 +1,75 @@
-import React, {useState} from 'react'
+import React, {useCallback, useState} from 'react'
 import {observer} from 'mobx-react'
 import {Button, useStellarNetwork} from '@stellar-expert/ui-framework'
 import accountLedgerData from '../../../state/ledger-data/account-ledger-data'
 import {requestFriendbotFunding} from '../../../util/horizon-connector'
+import DialogView from '../../layout/dialog-view'
 import {confirmTransaction} from './wallet-tx-confirmation'
+import {ConfirmIntentionView} from './confirm-intention-view'
 import ActionLoaderView from './action-loader-view'
 import './wallet.scss'
 
 /**
  * @param {String} title
  * @param {String} action?
  * @param {Bool} disabled?
+ * @param {Object} transfer?
  * @param {Function} prepareTransaction
  * @param {Function} onFinalize
  * @param {Boolean} allowNonExisting?
  * @param {*} children
  * @constructor
  */
-function WalletOperationsWrapperView({title, action, disabled, prepareTransaction, onFinalize, allowNonExisting, children}) {
+function WalletOperationsWrapperView({title, action, disabled, transfer, prepareTransaction, onFinalize, allowNonExisting, children}) {
```

Comment:
Please move all the logic related to the transaction confirmation into the separate component. And bear in mind that transfer may be undefined because this wrapper is used not only for transfers/swaps.